### PR TITLE
Show cmake output when version probe fails

### PR DESCRIPTION
**Pull request by geertj**
### This improves debuggability for users when the cmake version probe fails.

The following changes are proposed:

- When CMake Tools starts up, it tries to get information about the installed cmake version by executing the `cmake` binary. If the cmake invocation fails with a non-zero exit status, this change logs any standard output and standard error output.

### CODE REVIEW

**Comment by geertj** [comment]
@microsoft-github-policy-service agree company="Amazon"

**Comment by geertj** [comment]
(Apologies if it seemed like I was trying to merge this. I had imagined that the "Rebase" button would rebase the PR not merge this into main.)

File: src/proc.ts
Comment (gcampbell-msft) [inline src/proc.ts:217]: I think that if we're going to inject newlines, it would be good to include tabs so that it looks good in the output window. Or, just let it print by itself, especially in the log.warning call. Our loggers add `[proc]` to the beginning of the message, or a tab to ensure it's clear. I would say line 217 is fine, but 218 we shouldn't put the newline in.

**Comment by gcampbell-msft** [comment]
@geertj Just a couple small changes, please also add a CHANGELOG update. Thanks! Overall this looks like a good change!

**Review by gcampbell-msft — CHANGES_REQUESTED** [review]
See other comments for small changes requested.

File: src/proc.ts
Comment (geertj) [reply src/proc.ts]: I was inserting that prefix because some of our error messages are multi-line. I'm not sure how common this is, so maybe we should just go with the simplest approach and let it print by itself per one of your suggestions:

```
[proc] The command: [...]/cmake-wrapper --version exited with code: 1
[proc] Command output on standard error:
> First line of multi-line error message.
> Second line of multi-line error message.
```

New output:
```
[proc] The command: [...]/cmake-wrapper --version exited with code: 1
[proc] Command output on standard error: First line of multi-line error message.
Second line of multi-line error message.
```

Happy to add a tab or other prefix to second and later lines you prefer.

Comment (gcampbell-msft) [reply src/proc.ts]: I think that if it's easy enough, it'd be best to add tabs to second or later lines. However, if you think this is risky, I'm okay without it. Approving now but feel free to make that change as well.

**Review by gcampbell-msft — APPROVED** [review]

File: src/proc.ts
Comment (geertj) [reply src/proc.ts]: That's an easy change, updated!

**Review by gcampbell-msft — APPROVED** [review]

**Pull request merged**
